lib/location/wifi/wifi_nrf_cloud_rest.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <modem/lte_lc.h>
+#include <cJSON.h>
+#include <cJSON_os.h>
+
+#include <logging/log.h>
+
+#include <cJSON.h>
+
+#include <net/rest_client.h>
+#include <net/nrf_cloud.h>
+
+#include "wifi_nrf_cloud_rest.h"
+
+LOG_MODULE_DECLARE(location, CONFIG_LOCATION_LOG_LEVEL);
+
+#define HOSTNAME CONFIG_LOCATION_METHOD_WIFI_SERVICE_NRF_CLOUD_HOSTNAME
+
+BUILD_ASSERT(sizeof(HOSTNAME) > 1, "Hostname must be configured");
+
+#define REQUEST_URL "/v1/location/wifi"
+
+#define HEADER_CONTENT_TYPE "Content-Type: application/json\r\n"
+
+#define HTTPS_PORT 443
+
+#define NRF_CLOUD_WIFI_POS_JSON_KEY_AP "accessPoints"
+
+static char jwt_buf[1024];
```

Comment:
You seem to have a define for the size already: LOCATION_METHOD_WIFI_REST_RECV_BUF_SIZE.
Btw. I am using only 600 bytes for the JWT buffer on the GNSS side and it seems to work so far. I copied this from one of @plskeggs 's PRs, but I did not seek to confirm if it absolutely safe to use only 600 bytes.

Reply:
There are other comments on this topic: https://github.com/nrfconnect/sdk-nrf/pull/5963#discussion_r741036119

Reply:
Let's handle this in that other comment. Solving this.